src/main/groovy/com/twcable/grabbit/spring/batch/repository/JcrGrabbitJobInstanceDao.groovy

```diff
@@ -257,10 +257,21 @@ class JcrGrabbitJobInstanceDao extends AbstractJcrDao implements GrabbitJobInsta
         if (!resolver) throw new IllegalArgumentException("resolver == null")
 
         final rootResource = resolver.getResource(JOB_INSTANCE_ROOT)
-        final nextId = (rootResource.children.asList().size() + 1) as Long
+
+        final lastInstance = rootResource?.children?.asList()?.max { Resource resource ->
+            final properties = resource.adaptTo(ValueMap)
+            properties[INSTANCE_ID] as Long
+        }
+        final lastInstanceProperties = lastInstance?.adaptTo(ValueMap)
+        Long nextId = lastInstanceProperties?.get(INSTANCE_ID) as Long
+        if (!nextId) {
+            nextId = 1
+        } else {
+            nextId += 1
+        }
```

Comment:
The way that's implemented, contiguous sequential integers is implied. However it's external interface doesn't require it. As long as the ordering is consistent, there's no reason why that's needed.  `getJobInstance("foo", 0, 2)` and `getJobInstance("foo", 2, 2)` works for [999, 234, 452, 453] as long as it's guaranteed that the first call will not overlap with the second.

FWIW, "contiguous sequential integers" is convenient is simple situations, but maintaining that contract in concurrent distributed scenarios is... "tricky."